Acd 843 (#63)
Actindo-AG/Documentation: Payments/Integration/01_ManageConnection.md

[!!Connections (Settings)](../UserInterface/08a_Connections.md)

# Manage connections to payment service providers (PSP)   

If you want to establish a new connection to a payment service provider (PSP), you can buy, depending on the payment service provider, the needed connection in the app store or license it directly at the payment service provider. By default, Actindo supports the following drivers:
- Adyen&trade;
- Amazon pay&trade;
- EBICS&trade;
- Stripe&trade;
- A dummy driver to simulate payments for testing purposes    

The data structure of the connection then provides exactly the data that the PSP needs to exchange data with the Actindo *Payments* module.

> In the following, the general settings to create, edit, 

Reviewer: ```suggestion
In the following, the procedures to create, edit, edit, or disable connections are described. 
```

Author: Done

> or disable connections are described. 

Reviewer: Zeilenumbruch löschen

Author: done

## Create a PSP connection

Create a connection to a certain payment provider. As soon you have established a connection, an initial synchronization will be triggered, which will transfer the entire data model of the connected payment service provider to Actindo. 

> 

Reviewer: Bitte "line spacing" prüfen 
https://actindo.atlassian.net/wiki/spaces/DOC/pages/111345694/Line+spacing

Author: Done

#### Prerequisites

> At least one connection has been installed.

Reviewer: ```suggestion
At least one connection has been established.
```
Alternativ: created. Ich meine, so haben wir es vorher in anderen Modulen geschrieben.

Author: Done

#### Procedure    

*Payments > Settings > Tab CONNECTIONS*   

![Connections](../../Assets/Screenshots/Payments/Settings/Connections.png "[Connections]")
1.  Click the ![Add](../../Assets/Icons/Plus01.png "[Add]") (Add) button in the bottom right corner.   

>     The view for creating a connection is displayed.   

Reviewer: Wir haben immer den "Namen" des Fenster verwendet, auch wenn es im jetzigen UI nicht stimmt, damit man den Bezug im UI auch hat. Also: The "Create connection" view, the "Edit connection" view...

>     ![Create connection](../../Assets/Screenshots/Payments/Settings/CreateConnection.png "[Create connection]")

Reviewer: ```suggestion

    ![Create connection](../../Assets/Screenshots/Payments/Settings/CreateConnection.png "[Create connection]")
```
Line spacing (s.o.)

Author: done

2.  Enter a name for the connection in the *Name* field.

> 3.  Click the *Driver* drop-down list and select the desired driver. 

Reviewer: Leerzeile zwischen den Schritten.

Author: done

   All already installed drivers are displayed in the list.   
   

>      > [Info] If you want to install additional drivers, select the **more...** option and buy a driver in the app store. You have to purchase drivers from the main account. You need to install them separately in each sandbox or account where you want to use it.

Author: schon mit Oliver besprochen

     The *Credentials* section is displayed below the drop-down list.

    ![Credentials](../../Assets/Screenshots/Payments/Settings/CreateConnectionCredentials.png "[Credentials]")

>      > [Info] Depending on the selected driver, the fields in the *Credentials* section differ. 

Reviewer: Leerzeile fehlt. (Wie gesagt, am Besten mit dem ESG Line spacing vergleichen und durchgängig prüfen).

Reviewer: Leerzeile fehlt noch hier vor dem Info-Box.

4. Enter the required data in the fields in the *Credentials* section. Use the information you received from the payment service provider to fill the credentials.   

> 5. If the PSP connection requires additional settings, open the *Settings* tab and add further settings.

Reviewer: Kann man das so machen, bevor man die Verbindung erstellt hat? Oder muss man erst Speichern, also Verbindung erstellen, und dann erst die Settings bearbeiten?

Author: Done

6. Click the [SAVE] button.   

>     The connection will be established. The *Checking credentials...* notice is displayed. After a few seconds, the *Loading data...* notice is displayed.

Reviewer: Im Sandbox kommt keine "Loading data" Nachricht vor, aber vielleicht liegt es daran, dass es keine Settings mit dem Example Driver gibt. Evtl. Stefan fragen/zeigen lassen.

     ![Checking credentials](../../Assets/Screenshots/Payments/Settings/CheckingCredentials.png "[Checking credentials]")

>     The view for creating connections is automatically closed when the connection has been established. The *Edit connection* view of the newly created connection is displayed. The *Settings* tab is selected.

Reviewer: Im Sandbox (Example driver) wird die (List of) Connections wieder angezeigt, nicht die Settings tab der erstellten Verbindung, aber am Besten Stefan fragen.

>       ![Credentials](../../Assets/Screenshots/Payments/Settings/EditConnectionSettings.png "[Credentials]")

Reviewer: ```suggestion
      ![Settings](../../Assets/Screenshots/Payments/Settings/EditConnectionSettings.png "[Settings]")
```

Author: Da steht kein Settings auf dem Screen

Reviewer: Der Reiter heißt Settings, daher würde ich Settings nennen.

Author: in diesem Fall geht es um die Credentials

    <!-----Stefan: stimmt das? Ich kann es nicht nachmachen--->

## Edit a PSP connection

> Edit a connection to adjust any changed credentials, the connection name or further settings. 

Reviewer: ```suggestion
Edit a connection to adjust any changed credentials, the connection name or further settings. The driver selection is read-only and cannot be edited.
```

Author: Verstehe ich nicht

#### Prerequisites

> At least one connection has been created, see [Create a connection](#create-a-connection).

Reviewer: ```suggestion
At least one connection has been created, see [Create a connection](#create-a-PSP-connection).
```
Der Header heißt Create a PSP connection, dann der Link muss dementsprechend heißen. Oder Link so lassen und Header Name ändern.

Author: Done, created

#### Procedure

*Payments > Settings > Tab CONNECTIONS*

![Connections](../../Assets/Screenshots/Payments/Settings/ChangeConnections.png "[Connections]")

> 1. Click the checkbox on the left to select the connection you want to edit.  

Reviewer: Wenn man auf eine Verbindung anklickt, wird auch die Edit connection angezeigt (also man braucht nicht unbedingt den Checkbox anzuklicken und auf die Edit button klicken. In vielen Modulen ist der Edit button (sogenannte Selection mode teilweise abgeschafft, denn Entwickler benutzen es nicht, sie klicken einfach darauf. In Fulfillment und Omni-Channel haben wir es schon rausgelassen zu diesem Zeitpunkt).

Author: Done

   The editing toolbar is displayed above the list of connections.      

> 2. Click the ![Edit](../../Assets/Icons/Edit01.png "[Edit]") button

Reviewer: ```suggestion
2. Click the ![Edit](../../Assets/Icons/Edit01.png "[Edit]") button.
```

Author: Gelöscht

>    

Reviewer: ```suggestion
   ```

    The *Edit connection* view is displayed.

    > [Info] Depending on the driver, the *Credentials* as well as the *Settings* tab are displayed in the *Edit connection* view.

    ![Connection credentials](../../Assets/Screenshots/Payments/Settings/EditConnectionCredentials.png "[Connection credentials]")

2. Edit the desired data of the connection in the corresponding fields in the *Credentials* tab.

3. If necessary, click the *Settings* tab and edit the desired settings of the connection in the corresponding fields.

4. Click the [SAVE] button.   
    The *Checking credentials...* notice is displayed.
    ![Checking credentials](../../Assets/Screenshots/Payments/Settings/CheckingCredentials.png "[Checking credentials]")

>     The *Edit connection* view is automatically closed when the changes have been saved. The *CONNECTIONS* view is displayed again.

Reviewer: ```suggestion
    The *Edit connection* view is automatically closed when the changes have been saved. The *Connections* view is displayed again.
```

Author: Done

## Disable a PSP connection
Once a connection has been established, you cannot delete it because of potential dependencies in the existing database. Nevertheless, you can disable the connection if you no longer use it. You can disable several active connections at a time.

#### Prerequisites

> - At least one connection has been established, see [Create a connection](#create-a-connection).

Reviewer: s.o. (Link - Header)

Author: Done

> - The connection is active.

Reviewer: At least one connection is active. 

Vgl. Disable prerequisites.

Author: Done

#### Procedure

*Payments > Settings > Tab CONNECTIONS*

![Connections](../../Assets/Screenshots/Payments/Settings/ChangeConnections.png "[Connections]")

1. Select the checkbox of the connection you want to disable in the list of connections.   
    The editing toolbar is displayed above the list of connections.

2. Click the [DISABLE] button in the editing toolbar.   
    The connection has been deactivated.

3. Click the ![Refresh](../../Assets/Icons/Refresh01.png "[Refresh]") (Refresh) button to update the list of connections.   
    The status of the disabled connection in the *Status* column has switched to **Inactive**.

## Enable a PSP connection

> Once a connection has been established, you can disable it if you no longer need it. If you have done this by mistake, you can enable the connection again. You can enable several active connections at once.

Reviewer: ```suggestion
Once a connection has been established, you can disable it if you no longer need it. If you have done this by mistake, you can enable the connection again. You can enable several inactive connections at once.
```

> Once a connection has been established, you can disable it if you no longer need it. If you have done this by mistake, you can enable the connection again. You can enable several active connections at once.

Reviewer: Ist die Verbindung direkt aktiv oder inaktiv, nachdem man sie erstellt hat? Wenn inaktiv, dann würde ich erst Enable erklären.

#### Prerequisites

> 

Reviewer: - At least one connection has been established, see [Create a connection](#create-a-PSP-connection).

Author: done

> - At least one connection has the status *Inactive*.

Reviewer: ```suggestion
- At least one connection is inactive.
```
Vgl. oben.

Author: - Done

#### Procedure

*Payments > Settings > Tab CONNECTIONS*

![Connections](../../Assets/Screenshots/Payments/Settings/ChangeConnections.png "[Connections]")

> <!----Stefan, kannst Du mir einen Screenshot machen?Eigentlich brauche ich nur einen grünen Enable button --->

Reviewer: Wir versuchen, die Screenshots so viel wie möglich wieder zu verwenden, daher wird auch der Toolbar nicht angezeigt (Style Guide: the menu bar is not included to the screenshot -> https://actindo.atlassian.net/wiki/spaces/DOC/pages/96796734/General#Screenshots). Sonst passt es auch nicht mit dem Pfad.  Vgl. Omni-Channel oder Fulfillment.

Author: Da steht was von Menu bar nicht tool bar

1. Select the checkbox of the connection you want to enable in the list of connections.   
    The editing toolbar is displayed above the list of connections.

2. Click the [ENABLE] button in the editing toolbar.   
    The connection has been activated.

3. Click the ![Refresh](../../Assets/Icons/Refresh01.png "[Refresh]") (Refresh) button to update the list of connections.   

>     The *Checking credentials...* notice is displayed. 

Reviewer: Stimmt das so hier?

Author: Ja, auch in Sandbox kommt die Meldung. Aber nur für ein paar Millisekunden

>     The status of the enabled connection in the *Status* column has switched to **Active**.

Reviewer: Ich würde beide Sätze in einer Zeile schreiben.

[!!Connections (Settings)](../UserInterface/08a_Connections.md)

# Manage connections to payment service providers (PSP)   

If you want to create a new connection to a payment service provider (PSP), you can buy, depending on the payment service provider, the needed connection in the app store or license it directly at the payment service provider. By default, Actindo supports the following drivers:   
- Adyen&trade; (available in the app store)
- Amazon pay&trade; (available in the app store)   
It works without using the Amazon channel connection, but the Amazon driver must be installed.
- Stripe&trade; (available in the app store)
- Paypal&trade;
- A dummy driver to simulate payments for testing purposes in a sandbox   

The data structure of the connection then provides exactly the data that the PSP needs to exchange data with the Actindo *Payments* module.

> In the following, the procedures to create, edit, or disable connections are described. 

Reviewer: ```suggestion
In the following, the procedures to create, edit, disable, and enable connections are described. 
```

Author: Done

## Create PSP connection

Create a connection to a certain payment provider. As soon you have created a connection, an initial synchronization will be triggered, which will transfer the entire data model of the connected payment service provider to Actindo. 

#### Prerequisites

> - The relevant drivers are installed both on the main account and &ndash; if needed &ndash; the sandboxes.

Reviewer: ```suggestion
The relevant drivers are installed both on the main account and &ndash; if needed &ndash; the sandboxes.
```
"If needed" würde ich einfach zwischen Kommas schreiben, da keine "wichtige" Info.
Wenn man eine separate Info hinzufügen möchte, sollte man aber, meines Wissens, em dash (https://www.scribbr.com/language-rules/dashes/) schreiben, ohne Leerzeichen dazwischen, denke ich.

Author: done

#### Procedure    

*Payments > Settings > Tab CONNECTIONS*   

![Connections](../../Assets/Screenshots/Payments/Settings/Settings.png "[Connections]")

1.  Click the ![Add](../../Assets/Icons/Plus01.png "[Add]") (Add) button in the bottom right corner.   

>     The *Connection* view is displayed.   

Reviewer: The *Create connection* view is displayed.

Vgl. Omni-Channel: 
https://github.com/Actindo-AG/Documentation/blob/main/Channels/Integration/01_ManageConnections.md

Author: Auf dem Screen steht "Connection"

Author: Steht zwar nicht auf dem Screen, aber wenn es der Einheitlichkeit dient

    ![Create connection](../../Assets/Screenshots/Payments/Settings/CreateConnection.png "[Create connection]")

2.  Enter a name for the connection in the *Name* field.

3.  Click the *Driver* drop-down list and select the desired driver. 
   All already installed drivers are displayed in the list.   
   

>      > [Info] If you want to install additional drivers, select the **more...** option and buy a driver in the app store. You have to purchase drivers from the main account. You need to install them separately in each sandbox or account where you want to use it.

Reviewer: ```suggestion
    > [Info] If you want to install additional drivers, select the **more...** option and buy a driver in the app store. You have to purchase drivers from the main account. You need to install them separately in each sandbox or account where you want to use it.
```
Einrückung (zu weit eingerückt)

Author: done

     The *Credentials* section is displayed below the drop-down list.

    ![Credentials](../../Assets/Screenshots/Payments/Settings/CreateConnectionCredentials.png "[Credentials]")

>      > [Info] Depending on the selected driver, the fields in the *Credentials* section differ. 

Reviewer: Leerzeile vor dem Info fehlt

Author: done

5. Click the [SAVE] button.   
    The connection will be created. Depending on the driver, the *Checking credentials...* notice is displayed. After a few seconds, the *Loading data...* notice is displayed.

     ![Checking credentials](../../Assets/Screenshots/Payments/Settings/CheckingCredentials.png "[Checking credentials]")

    The data are synchronized between Actindo and the payment service provider, if a synchronization is necessary. The view for creating connections is automatically closed when the connection has been created and synchronized. The *Edit connection* view of the newly created connection is displayed. The *Settings* tab is selected.
      ![Credentials](../../Assets/Screenshots/Payments/Settings/EditConnectionSettings.png "[Credentials]")
    <!---Stefan: stimmt das? Ich kann es nicht nachmachen, wenn es Verbindungen sind, die nicht direkt aktiv sind, gibt es dann buttons mit denen man aktivieren, synchronisieren kann?-->

## Edit PSP connection

Edit a connection to adjust any changed credentials, the connection name or further settings. 

#### Prerequisites

> At least one connection has been created, see [Create a connection](#create-a-psp-connection).

Reviewer: ```suggestion
At least one connection has been created, see [Create PSP connection](#create-psp-connection).
```
Oder [Create a PSP connection](#create-a-psp-connection) und Header dementsprechend umbenennen

Author: done

#### Procedure

*Payments > Settings > Tab CONNECTIONS*

![Connections](../../Assets/Screenshots/Payments/Settings/Settings.png "[Connections]")

1. Select the checkbox of the driver to select the connection you want to edit.  
The editing toolbar is displayed above the list of connections.

2. Click the ![Edit](../../Assets/Icons/Edit01.png "[Edit") (Edit) button.   

>     The *"Driver name" connection* view is displayed.

Reviewer: The *Edit connection* view is displayed.

Vgl. Omni-Channel: 
https://github.com/Actindo-AG/Documentation/blob/main/Channels/Integration/01_ManageConnections.md

Author: done

    > [Info] Depending on the driver, the *Credentials* as well as the *Settings* tab are displayed in the view.

    ![Credentials](../../Assets/Screenshots/Payments/Settings/EditConnectionCredentials.png "[Connection credentials]")

> 2. Edit the desired data of the connection in the corresponding fields in the *Credentials* tab.   

Reviewer: Achtung: Nummerierung -> 2. hier sollte 3. sein usw.

Author: done

3. If necessary, click the *Settings* tab and edit the desired settings of the connection in the corresponding fields.

4. Click the [SAVE] button.   

>     The *Checking credentials...* notice is displayed.

Reviewer: Einrückung

Author: Done

>     ![Checking credentials](../../Assets/Screenshots/Payments/Settings/CheckingCredentials.png "[Checking credentials]")

Reviewer: Leerzeilen vor und nach dem Screenshot (das hat eine Auswirkung in GitHub, also die Leerzeilen fehlen auch auf der Webseite)

Author: done

>     The *"Driver name" connection* view is automatically closed when the changes have been saved. The *Connections* view is displayed again.

Reviewer: s.o.

Author: done

## Disable PSP connection

Once a connection has been and is in use, you cannot delete it because of potential dependencies in the existing database. Nevertheless, you can disable the connection if you no longer use it. You can disable several active connections at a time.

#### Prerequisites

> - At least one connection is active, see [Create a connection](#create-a-psp-connection).

Reviewer: ```suggestion
At least one connection is active, see [Create PSP connection](#create-psp-connection).
```

Author: Done

#### Procedure

*Payments > Settings > Tab CONNECTIONS*

![Connections](../../Assets/Screenshots/Payments/Settings/Settings.png "[Connections]")

1. Select the checkbox of the connection you want to disable in the list of connections.   
   The editing toolbar is displayed above the list of connections.

2. Click the [DISABLE] button in the editing toolbar.   
   The connection has been deactivated.

3. Click the ![Refresh](../../Assets/Icons/Refresh01.png "[Refresh]") (Refresh) button to update the list of connections.   
    The status of the disabled connection in the *Status* column has switched to **Inactive**.

## Enable PSP connection

Once a connection has been created, you can disable it if you no longer need it. If you have done this by mistake, you can enable the connection again. You can enable several inactive connections at once.

#### Prerequisites

> - At least one connection is *Inactive*.

Reviewer: ```suggestion
At least one connection is **Inactive**.
```
Kein "bullet point", wenn nur ein Prerequisite. Vgl. unten **Active**.

Author: done

#### Procedure

*Payments > Settings > Tab CONNECTIONS*

> ![Connections](../../Assets/Screenshots/Payments/Settings/settings.png "[Connections]")

Reviewer: Check bitte Link zu Screenshot. Es wird nicht angezeigt.

Author: Done

> 

Reviewer: Eine Leerzeile zu viel

Author: done

1. Select the checkbox of the connection you want to enable in the list of connections.   
    The editing toolbar is displayed above the list of connections.

2. Click the [ENABLE] button in the editing toolbar.   
    The connection has been activated.

3. Click the ![Refresh](../../Assets/Icons/Refresh01.png "[Refresh]") (Refresh) button to update the list of connections.   

>         The status of the enabled connection in the *Status* column has switched to **Active**.

Reviewer: Zu weit eingerückt. Prüfe bitte die anderen Schritte auch.

Author: done
